### PR TITLE
User can select a project

### PR DESCRIPTION
**Pull request by Boyul-Kim**
6997f5f8cb0a54d715934bfeebe3d650

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -51,6 +51,40 @@ app.get('/api/home/:projectId', (req, res, next) => {
...
+    where "projectId" = ($1)
+  `;
+
+  const param = [projectId];
+
+  db.query(sql, param)
+    .then(result => {
+      if (!result.rows) {
+        throw new ClientError(404, `cannot find projectId ${projectId}`);
+      }
```
Comment (thebearingedge) [inline server/index.js]: `result.rows` is an array and arrays are truthy. Also, an empty list is not a `404`. "All" actually means "zero or more".

File: client/pages/project.jsx [hunk 2]
```diff
@@ -4,19 +4,86 @@ export default class Project extends React.Component {
...
+            </div>
+
+          </form>
+
+          <select className="custom-select custom-select-sm mb-3">
+            <option defaultValue>Filter</option>
+            <option value="1">One</option>
+            <option value="2">Two</option>
+            <option value="3">Three</option>
+          </select>
```
Comment (thebearingedge) [inline client/pages/project.jsx]: You can leave this out until the next feature. Including extra code gives your reviewer more work. If you had to understand code you were given, would you prefer more code or less code?

File: client/pages/project.jsx [hunk 3]
```diff
@@ -4,19 +4,86 @@ export default class Project extends React.Component {
...
+          }
+          <form>
+            <div className="form-row mb-2">
+              <div className="col-10">
+                <input type="text" className="form-control vw-90" id="searchBar" placeholder="Search task" />
+              </div>
+              <button type="submit" className="btn btn-primary mb-2">Go!</button>
+            </div>
+
+          </form>
```
Comment (thebearingedge) [inline client/pages/project.jsx]: You can leave this out until the next feature. Including extra code gives your reviewer more work. If you had to understand code you were given, would you prefer more code or less code?

File: client/pages/project.jsx [hunk 4]
```diff
@@ -4,19 +4,86 @@ export default class Project extends React.Component {
...
   }
 
   render() {
+
     return (
-      <div>test</div>
+      <div>
+        <nav className="navbar navbar-expand-xl navbar-color navbar-height">
+          <i className="fas fa-bars white fa-2x"></i>
+        </nav>
```
Comment (thebearingedge) [inline client/pages/project.jsx]: You can leave this out until it does something. Including extra code gives your reviewer more work. If you had to understand code you were given, would you prefer more code or less code?

File: client/pages/project.jsx [hunk 5]
```diff
@@ -4,19 +4,86 @@ export default class Project extends React.Component {
...
+      details: []
     };
   }
 
   componentDidMount() {
     fetch(`/api/home/${this.props.projectId}`)
       .then(res => res.json())
       .then(result => this.setState({ project: result }));
+
+    fetch(`/api/projects//${this.props.projectId}`)
```
Comment (thebearingedge) [inline client/pages/project.jsx]: There is an extra slash here.

File: client/pages/project.jsx [hunk 6]
```diff
@@ -4,19 +4,86 @@ export default class Project extends React.Component {
...
-      project: null
+      project: null,
+      details: []
     };
   }
 
   componentDidMount() {
     fetch(`/api/home/${this.props.projectId}`)
       .then(res => res.json())
       .then(result => this.setState({ project: result }));
```
Comment (thebearingedge) [inline client/pages/project.jsx]: What is this for?

File: client/pages/project.jsx [hunk 7]
```diff
@@ -4,19 +4,86 @@ export default class Project extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      project: null
+      project: null,
```
Comment (thebearingedge) [inline client/pages/project.jsx]: What is this for?

File: client/pages/project.jsx [hunk 8]
```diff
@@ -4,19 +4,86 @@ export default class Project extends React.Component {
...
+            <option value="1">One</option>
+            <option value="2">Two</option>
+            <option value="3">Three</option>
+          </select>
+
+          <div className="card">
+            <div className="card-body">
+              <h4>Current Tasks</h4>
+              {
+                this.state.details.map(detail => (
```
Comment (thebearingedge) [inline client/pages/project.jsx]: Your list of current tasks should probably be in an array named `this.state.tasks`. Try to use the same words to describe things when applicable.

File: client/pages/project.jsx [hunk 9]
```diff
@@ -4,19 +4,86 @@ export default class Project extends React.Component {
...
+                    }
+                  </div>
+                ))
+              }
+            </div>
+          </div>
+
+          <div>
+            <i className="fas fa-plus fa-2x row d-flex justify-content-center homepage-font mt-3"></i>
+          </div>
```
Comment (thebearingedge) [inline client/pages/project.jsx]: You can add this when it actually does something.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Ok good progress. Please see my notes. Also, please take more time to make your task list match your figma.

**Comment by Boyul-Kim** [comment]
14e08218944edc6fbe6e1ea97e4b0c2a

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**